src/NServiceBus.RavenDB/SessionManagement/RavenDbStorageSession.cs

```diff
@@ -12,26 +12,31 @@ protected override void Setup(FeatureConfigurationContext context)
             context.Container.ConfigureComponent<RavenDBSynchronizedStorageAdapter>(DependencyLifecycle.SingleInstance);
             context.Container.ConfigureComponent<RavenDBSynchronizedStorage>(DependencyLifecycle.SingleInstance);
 
-            IOpenRavenSessionsInPipeline sessionCreator;
-
             // Check to see if the user provided us with a shared session to work with before we go and create our own to inject into the pipeline
             var getAsyncSessionFunc = context.Settings.GetOrDefault<Func<IDictionary<string, string>, IAsyncDocumentSession>>(RavenDbSettingsExtensions.SharedAsyncSessionSettingsKey);
 
             if (getAsyncSessionFunc != null)
             {
-                sessionCreator = new OpenRavenSessionByCustomDelegate(getAsyncSessionFunc);
+                IOpenRavenSessionsInPipeline sessionCreator = new OpenRavenSessionByCustomDelegate(getAsyncSessionFunc);
+                context.Container.RegisterSingleton(sessionCreator);
```

Comment:
Now `sessionCreator` is registered in the container only in this branch. Previously it was registered always. Should we keep the behavior the same?

Reply:
https://github.com/Particular/NServiceBus.RavenDB/pull/451/files#diff-183ab1023ea69e1b17a407a0d4085582R25 also registers an `IOpenRavenSessionsInPipeline`  (sessionCreater is the same type) so there should always be a session creator registered?